source/Engine.cpp

```diff
@@ -1252,6 +1255,9 @@ void Engine::MoveShip(const shared_ptr<Ship> &ship)
 		// self-destruct.
 		if(ship->IsDestroyed())
 			eventQueue.emplace_back(nullptr, ship, ShipEvent::DESTROY);
+		// Record that the ship was able to flee (by landing).
+		else
```

Comment:
This should not be an `else` unless you want
 - every random NPC that lands to throw a FLEE event
 - every random NPC that is out of system for 1000 frames to throw a FLEE event
 - every fighter that docks with its carrier to throw a FLEE event

Reply:
For drones/fighters that dock, they shouldn't throw a FLEE until the carrier throws a FLEE. Will change.

Other than that particular case I believe being removed from the game while still alive is a valid "flee" event since you won't be seeing the same ship again. But this will probably be a performance bottleneck at some point  `O(flees*missions*npcs)`. Ship should probably have a variable that indicates which events it can trigger.

Reply:
In #2990 I used two ship methods (1 new):
1) `HasLanded() - returns a property that gets set upon landing, which means fighters and disappearances can't throw the event
2) `IsSpecial()` - if the ship isn't special, a landing event is not of any importance, so don't throw it then either

Reply:
Then if the ship isn't special, it should not throw a DESTROY either.

Reply:
I believe there are rep hits associated with destruction events, so I wouldn't go that far